src/components/TopBar.js

```diff
@@ -397,70 +296,41 @@ export default _.flow(
     ])
   }
 
-  buildUserSection() {
-    const { authState: { profile: { firstName = 'Loading...', lastName = '' } } } = this.props
-    const { userMenuOpen } = this.state
-
+  buildDropDownSection(titleIcon, title, onClick, menuOpen, subItems) {
     return h(Collapse, {
       defaultHidden: true,
       showIcon: false,
       animate: true,
       expandTitle: true,
-      style: styles.nav.profile(false),
       buttonStyle: { marginBottom: 0 },
       title: [
         h(Clickable, {
-          style: { ...styles.nav.item, ...styles.nav.profile(userMenuOpen) },
-          hover: styles.nav.profile(true),
-          onClick: () => this.setState({ userMenuOpen: !userMenuOpen })
+          style: styles.nav.item,
+          hover: { backgroundColor: colors.gray[3] },
+          onClick
         }, [
           div({ style: styles.nav.icon }, [
-            profilePic({ size: 32 })
-          ]),
-          div({ style: { whiteSpace: 'nowrap', overflow: 'hidden', textOverflow: 'ellipsis' } }, [
-            `${firstName} ${lastName}`
+            (titleIcon === undefined) ? profilePic({ size: 32 }) : icon(titleIcon, {
```

Comment:
Rather than having this as a special case, what about having the icon or pic be passed in as part of the title contents?